staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -88,6 +101,42 @@ var (
 		},
 		[]string{priorityLevel, flowSchema},
 	)
+	windowedRequestCountWatermarks = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_watermark",
+			Help:      "Minimum number of requests queued, executing in recent windows",
+		},
+		[]string{priorityLevel, phase, mark, lag},
+	)
+	windowedRequestCountAvg = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_avg",
+			Help:      "Average number of requests queued, executing over recent windows",
+		},
+		[]string{priorityLevel, phase},
+	)
+	windowedRequestCountStddev = compbasemetrics.NewGaugeVec(
+		&compbasemetrics.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_stddev",
+			Help:      "Standard deviation of number of requests queued, executing over recent windows",
+		},
+		[]string{priorityLevel, phase},
+	)
+	windowedRequestCountIntegrals = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "windowed_request_count_integrals",
+			Help:      "Integrals of powers of number of requests queued, executing since startup",
+		},
+		[]string{priorityLevel, phase, power},
```

Comment:
power probably shouldn't be a dimension.  A single metric should be meaningful when aggregated across dimensions.  Do the units of the different powers match?

Reply:
Of course the units do not match.  In the application at hand, the units are duration, requests*duration, and requests^2*duration, respectively.